yb-voyager/cmd/endMigrationCommand.go

```diff
@@ -286,6 +287,36 @@ func saveMigrationReportsFn(msr *metadb.MigrationStatusRecord) {
 	}
 }
 
+func saveSchemaOptimizationReport() {
+	optimizationReportGlobPath := filepath.Join(backupDir, "reports", fmt.Sprintf("%s.*", SCHEMA_OPTIMIZATION_REPORT_FILE_NAME))
+	alreadyBackedUp := utils.FileOrFolderExistsWithGlobPattern(optimizationReportGlobPath)
+	if alreadyBackedUp {
+		utils.PrintAndLog("schema optimization report is already present at %q", optimizationReportGlobPath)
+		return
+	}
+	utils.PrintAndLog("saving schema optimization report...")
+
+	files, err := os.ReadDir(filepath.Join(exportDir, "reports"))
+	if err != nil {
+		utils.ErrExit("reading reports directory: %w", err)
+	}
+	for _, file := range files {
```

Comment:
we know the exact file location. Can we not directly check if it exists or not. Why get the files in the reports, and then do a prefix match?